### PR TITLE
Remove use of deprecated APIs being removed in Kubernetes v1.22

### PR DESCRIPTION
**Pull request by simonpasquier**
* [ ] I added CHANGELOG entry for this change.
* [X] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -18,18 +18,18 @@ require (
...
 	k8s.io/klog/v2 v2.8.0
 	k8s.io/kube-aggregator v0.20.0
 	k8s.io/metrics v0.19.4
 )
 
 replace (
-	k8s.io/api => k8s.io/api v0.21.0
-	k8s.io/apimachinery => k8s.io/apimachinery v0.21.0
-	k8s.io/client-go => k8s.io/client-go v0.21.0
+	k8s.io/api => k8s.io/api v0.21.2
```
Comment (prashbnair) [inline go.mod:32]: I think we need the replace only for the client-go library.

Comment (simonpasquier) [reply go.mod]: correct! updated.

**Pull request merged**